src/colorize.cr

```diff
@@ -323,23 +351,47 @@ struct Colorize::Object(T)
     back color
   end
 
+  # Enables or disables colors and text decoration on this object.
   def toggle(flag)
     @enabled = !!flag
     self
   end
 
+  # Appends this object colored and with text decoration to *io*.
   def to_s(io : IO) : Nil
     surround(io) do
       io << @object
     end
   end
 
+  # Inspects this object and makes the ANSI escape codes visible.
   def inspect(io : IO) : Nil
     surround(io) do
       @object.inspect(io)
     end
   end
 
+  # Surrounds *io* by the ANSI escape codes and lets you build colored strings:
+  #
+  # ```
+  # require "colorize"
+  #
+  # io = IO::Memory.new
+  #
+  # with_color.red.surround(io) do
+  #   io << "colorful"
+  #   with_color.green.bold.surround(io) do
+  #     io << " hello "
+  #   end
+  #   with_color.blue.surround(io) do
+  #     io << "world"
+  #   end
+  #   io << " string"
+  # end
+  #
+  # io.to_s # => "colorful hello world string"
```

Comment:
Welp this probably is an interesting breakage case for https://github.com/maiha/crystal-examples
But just mentioning this, I don't have any direct suggestion.

Reply:
That's right. Will need to address that.